File: ArchitectureComponents/GithubArchitectureComponents/app/src/main/java/com/dodemy/githubarchitecturecomponents/di/module/ViewModelModule.java
package com.dodemy.githubarchitecturecomponents.di.module;



import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

import com.dodemy.githubarchitecturecomponents.di.key.ViewModelKey;
import com.dodemy.githubarchitecturecomponents.view_models.FactoryViewModel;
import com.dodemy.githubarchitecturecomponents.view_models.UserProfileViewModel;

import dagger.Binds;
import dagger.Module;
import dagger.multibindings.IntoMap;



@Module
public abstract class ViewModelModule {

    @Binds
    @IntoMap
    @ViewModelKey(UserProfileViewModel.class)
    abstract ViewModel bindUserProfileViewModel(UserProfileViewModel repoViewModel);

    @Binds
    abstract ViewModelProvider.Factory bindViewModelFactory(FactoryViewModel factory);
}

File: Kotlin/LegoCatalogMVVMRoomPaging/app/src/main/java/com/dodemy/legocatalogmvvmroompaging/di/ViewModelModule.kt
package com.dodemy.legocatalogmvvmroompaging.di

import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import com.dodemy.legocatalogmvvmroompaging.legoset.ui.LegoSetViewModel
import com.dodemy.legocatalogmvvmroompaging.legoset.ui.LegoSetsViewModel
import com.dodemy.legocatalogmvvmroompaging.legotheme.ui.LegoThemeViewModel

import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap

@Suppress("unused")
@Module
abstract class ViewModelModule {
    @Binds
    @IntoMap
    @ViewModelKey(LegoThemeViewModel::class)
    abstract fun bindThemeViewModel(viewModel: LegoThemeViewModel): ViewModel

    @Binds
    @IntoMap
    @ViewModelKey(LegoSetsViewModel::class)
    abstract fun bindLegoSetsViewModel(viewModel: LegoSetsViewModel): ViewModel

    @Binds
    @IntoMap
    @ViewModelKey(LegoSetViewModel::class)
    abstract fun bindLegoSetViewModel(viewModel: LegoSetViewModel): ViewModel

    @Binds
    abstract fun bindViewModelFactory(factory: ViewModelFactory): ViewModelProvider.Factory
}
